pkg/proxy/server.go

```diff
@@ -53,8 +50,7 @@ func NewServer(ctx context.Context, c CompletedConfig) (*Server, error) {
 	s := &Server{
 		CompletedConfig: c,
 	}
-	rootShardConfigInformerConfig := kcpclienthelper.SetCluster(restclient.CopyConfig(s.CompletedConfig.RootShardConfig), tenancyv1alpha1.RootCluster)
-	rootShardConfigInformerClient, err := kcpclient.NewForConfig(rootShardConfigInformerConfig)
+	rootShardConfigInformerClient, err := kcpclientset.NewForConfig(s.CompletedConfig.RootShardConfig)
```

Comment:
@p0lyn0mial please check

Reply:
How does it know about the `root` cluster? I think it is implicit, it relies on the assumption that an admin will set a valid kubeconfig context.

I'd prefer to explicitly set it to the `root` cluster and then to use `NewSharedScopedInformerFactory`, wdyt?

Reply:
If we want a cluster-scoped LIST+WATCH, yes, 100% you are correct @p0